diff --git a/hw/vendor/openhwgroup_cv32e40p/bhv/cv32e40p_tb_wrapper.sv b/hw/vendor/openhwgroup_cv32e40p/bhv/cv32e40p_tb_wrapper.sv
index bed802c..69d6f7e 100644
--- a/hw/vendor/openhwgroup_cv32e40p/bhv/cv32e40p_tb_wrapper.sv
+++ b/hw/vendor/openhwgroup_cv32e40p/bhv/cv32e40p_tb_wrapper.sv
@@ -15,7 +15,9 @@
 `include "cv32e40p_prefetch_controller_sva.sv"
 `endif
 
+`ifdef CV32E40P_CORE_LOG
 `include "cv32e40p_core_log.sv"
+`endif
 
 `ifdef CV32E40P_APU_TRACE
 `include "cv32e40p_apu_tracer.sv"
@@ -95,7 +97,7 @@ module cv32e40p_tb_wrapper
 
 `endif  // CV32E40P_ASSERT_ON
 
-`ifndef SYNTHESIS
+`ifndef CV32E40P_CORE_LOG
   cv32e40p_core_log #(
       .PULP_XPULP      (PULP_XPULP),
       .PULP_CLUSTER    (PULP_CLUSTER),
@@ -109,7 +111,7 @@ module cv32e40p_tb_wrapper
       .hart_id_i             (cv32e40p_wrapper_i.core_i.hart_id_i),
       .pc_id_i               (cv32e40p_wrapper_i.core_i.pc_id)
   );
-`endif  // SYNTHESIS
+`endif  // CV32E40P_CORE_LOG
 
 `ifdef CV32E40P_APU_TRACE
   cv32e40p_apu_tracer apu_tracer_i (
