diff --git a/src/dm_csrs.sv b/src/dm_csrs.sv
index 59d1c90..2338028 100644
--- a/src/dm_csrs.sv
+++ b/src/dm_csrs.sv
@@ -547,7 +547,7 @@ module dm_csrs #(
     if (!dmcontrol_q.resumereq && dmcontrol_d.resumereq) begin
       clear_resumeack_o = 1'b1;
     end
-    if (dmcontrol_q.resumereq && resumeack_i) begin
+    if (dmcontrol_q.resumereq && resumeack_i[selected_hart]) begin
       dmcontrol_d.resumereq = 1'b0;
     end
     // static values for dcsr
diff --git a/src/dm_mem.sv b/src/dm_mem.sv
index 9ff3c86..8f97bec 100755
--- a/src/dm_mem.sv
+++ b/src/dm_mem.sv
@@ -535,8 +535,8 @@ module dm_mem #(
 
   always_ff @(posedge clk_i or negedge rst_ni) begin
     if (!rst_ni) begin
-      halted_q   <= 1'b0;
-      resuming_q <= 1'b0;
+        halted_q   <= '0;
+        resuming_q <= '0;
     end else begin
       halted_q   <= SelectableHarts & halted_d;
       resuming_q <= SelectableHarts & resuming_d;
