diff --git a/cve2.core b/cve2.core
index ec43f9c..08e1437 100644
--- a/cve2.core
+++ b/cve2.core
@@ -148,7 +148,7 @@ targets:
       - tool_veriblelint ? (files_lint_verible)
       - files_rtl
       - target_sim? (files_clk_gate)
-      - target_sim-opt? (files_clk_gate)
+      - target_sim_opt? (files_clk_gate)
     toplevel: ibex_core
     parameters:
       - tool_vivado ? (FPGA_XILINX=true)
