diff --git a/rtl/cve2_ex_block.sv b/rtl/cve2_ex_block.sv
index f84d990f..710fb26e 100644
--- a/rtl/cve2_ex_block.sv
+++ b/rtl/cve2_ex_block.sv
@@ -166,6 +166,13 @@ module cve2_ex_block #(
       .valid_o           (multdiv_valid),
       .multdiv_result_o  (multdiv_result)
     );
+  end else begin : gen_no_multdiv
+    assign multdiv_result = '0;
+    assign multdiv_alu_operand_a = '0;
+    assign multdiv_alu_operand_b = '0;
+    assign multdiv_valid = '0;
+    assign multdiv_imd_val_d = {'0, '0};
+    assign multdiv_imd_val_we = '0;
   end
 
   // Multiplier/divider may require multiple cycles. The ALU output is valid in the same cycle
