diff --git a/src/fpu_ss.sv b/src/fpu_ss.sv
index c3e3be3..1c197ed 100644
--- a/src/fpu_ss.sv
+++ b/src/fpu_ss.sv
@@ -95,13 +95,8 @@ module fpu_ss
 );
 
 // predecoder parameter
-`ifdef PULP_ZFINX_DEF
-  localparam int unsigned NUM_INSTR                   = fpu_ss_prd_zfinx_pkg::NumInstr;
-  localparam offload_instr_t OFFLOAD_INSTR[NUM_INSTR] = fpu_ss_prd_zfinx_pkg::OffloadInstr;
-`else
-  localparam int unsigned NUM_INSTR                   = fpu_ss_prd_f_pkg::NumInstr;
-  localparam offload_instr_t OFFLOAD_INSTR[NUM_INSTR] = fpu_ss_prd_f_pkg::OffloadInstr;
-`endif
+  localparam int unsigned NUM_INSTR                   = PULP_ZFINX ? fpu_ss_prd_zfinx_pkg::NumInstr: fpu_ss_prd_f_pkg::NumInstr;
+  localparam offload_instr_t OFFLOAD_INSTR[NUM_INSTR] = PULP_ZFINX ? fpu_ss_prd_zfinx_pkg::OffloadInstr : fpu_ss_prd_f_pkg::OffloadInstr;
 
   // compressed predecoder signals
   comp_prd_req_t                                  comp_prd_req;
