diff --git a/./tech_cells_generic/src/rtl/tc_clk.sv b/./x-heep/hw/vendor/pulp_platform_tech_cells_generic/src/rtl/tc_clk.sv
index e6aa1fd..24fedde 100644
--- a/./tech_cells_generic/src/rtl/tc_clk.sv
+++ b/./x-heep/hw/vendor/pulp_platform_tech_cells_generic/src/rtl/tc_clk.sv
@@ -8,6 +8,8 @@
 // CONDITIONS OF ANY KIND, either express or implied. See the License for the
 // specific language governing permissions and limitations under the License.
 
+/* verilator lint_off UNUSED */
+
 module tc_clk_and2 (
   input  logic clk0_i,
   input  logic clk1_i,
