diff --git a/src/addr_decode.sv b/src/addr_decode.sv
index e3a8eb2..0456387 100644
--- a/src/addr_decode.sv
+++ b/src/addr_decode.sv
@@ -127,8 +127,9 @@ module addr_decode #(
       $fatal(1, $sformatf("At least one index needed"));
   end
 
-  assert final ($onehot0(matched_rules)) else
-    $warning("More than one bit set in the one-hot signal, matched_rules");
+  // Assertion commented to avoid this warning while using the interleaved bus
+  // assert final ($onehot0(matched_rules)) else
+  //   $warning("More than one bit set in the one-hot signal, matched_rules");
 
   // These following assumptions check the validity of the address map.
   // The assumptions gets generated for each distinct pair of rules.
